### PR TITLE
Revert PR #4068 and use pagination_total optimization with decorators

### PR DESCRIPTION
**Pull request by rafaelsales**
The PR https://github.com/activeadmin/activeadmin/pull/3848 introduced an optimization that prevents COUNT queries when using `index pagination_total: false`.

I noticed that when using the macro `decorate_with`, the optimization didn't work and the count query across the entire database was made. That is because the AA default collection decorator didn't implement `offset`. I believe the PR https://github.com/activeadmin/activeadmin/pull/4068 made the wrong fix for this

- [x] Fix: revert https://github.com/activeadmin/activeadmin/pull/4068 and make the collection decorator delegate `offset` to AR relation
- [x] Remove ORDER BY from count subquery. Queries like `SELECT COUNT(*) FROM (SELECT DISTINCT resources.* FROM resources ORDER BY resources.created_at DESC LIMIT 1 OFFSET 30) subquery_for_count` are very inefficient

### CODE REVIEW

**Comment by rafaelsales** [comment]
@varyonic fixed!

**Comment by zorab47** [comment]
Any update on my question for the `offset` check @rafaelsales?

Is there a way to time out queries in a multi-database way to only allocate a certain amount of time for count queries?

File: lib/active_admin/views/components/paginated_collection.rb [hunk 1]
```diff
@@ -96,13 +96,15 @@ def build_pagination
...
         options[:param_name] = @param_name if @param_name
 
-        if !@display_total && collection.respond_to?(:offset)
+        if !@display_total
           # The #paginate method in kaminari will query the resource with a
           # count(*) to determine how many pages there should be unless
           # you pass in the :total_pages option. We issue a query to determine
           # if there is another page or not, but the limit/offset make this
           # query fast.
-          offset = collection.offset(collection.current_page * @per_page.to_i).limit(1).count
```
[the remark this answers is not in the document]
Comment (rafaelsales) [reply lib/active_admin/views/components/paginated_collection.rb]: It was never required (https://github.com/activeadmin/activeadmin/pull/4068) - it was just a different way to optimize.

**Comment by rafaelsales** [comment]
@zorab47 
> Is there a way to time out queries in a multi-database way to only allocate a certain amount of time for count queries?
Maybe... but this is certainly out of scope :)

I really wanted to get rid of the monkey patch from my apps 😭

**Comment by deivid-rodriguez** [comment]
@rafaelsales Sorry for missing this PR. We have (at least partially) fixed this in https://github.com/activeadmin/activeadmin/pull/5811. I should've included you in the author's list because the solution originates from this PR, sorry about that.

This PR includes some extra changes that are probably further optimizations. Would you like to rebase this PR so we can also add those improvements?

Thanks!

**Comment by deivid-rodriguez** [comment]
Rebased as #6911.

**Pull request closed without merging**